components/elm/src/biogeophys/SoilMoistStressMod.F90

```diff
@@ -401,14 +404,14 @@ subroutine calc_root_moist_stress_clm45default(bounds, &
                endif
 
                !use floodf to change root water uptake
-               if (h2osfc(1) .ge. 0._r8 .and. h2osfc(1) .le. 1000.0_r8) then
-                     floodf(veg_pp%itype(p))=1-0.001*h2osfc(1)
-               elseif(h2osfc(1) .gt. 1000._r8) then
-                     floodf(veg_pp%itype(p))=0.0_r8
-               elseif(h2osfc(1) .lt. 0._r8) then
-                     floodf(veg_pp%itype(p))=1.0_r8                       
+               if (h2osfc(1) .gt. 0._r8 .and. h2osfc(1) .lt. htop(p)*1000) then
+                     floodf(1)=(htop(p)*1000-h2osfc(1))/(htop(p)*1000)
+               elseif(h2osfc(1) .ge. htop(p)*1000) then
+                     floodf(1)=0.0_r8
+               elseif(h2osfc(1) .le. 0._r8) then
+                     floodf(1)=1.0_r8                       
```

Comment:
I'm trying to avoid hard-coding column numbers, so maybe change these from (1) to (c) if they are column defined